app/models/graders/junit_grader.rb

```diff
@@ -50,6 +57,21 @@ def import_data_schema
     "junit_import_schema"
   end
 
+  def generate_grading_job_json(sub)
+    ans = {}
+    ans["key"] = JSON.generate({ grade_id: self.grade_for(sub).id })
```

Comment:
Were you going to incorporate the server-response URL into this key, or was that only going to be on the Orca side?  I.e., do we want something like `ans["key"] = JSON.generate({ grade_id, response_url})`, or do we want `ans["key"]` and `ans["response_url"]` as two separate fields?  Basically, is Orca going to be responsible for enforcing that the primary-key for its jobs is comprised of two fields?  (I can see arguments both ways, here; I'm not sure which design I think is best.)

Reply:
Counterargument: the response URL is meaningful in plaintext, and the key so far is opaque to Orca and meaningful _only_ to the originating server.  So it makes sense for Orca to combine those two into something it can use as a primary key, and it makes less sense to design Orca to require that all of its customers provide a key that is somehow both globally unique and also only semi-structured.